specification/purview/resource-manager/Microsoft.Purview/preview/2023-05-01-preview/purview.json

```diff
@@ -0,0 +1,2980 @@
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/accountName"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "in": "body",
+            "name": "featureRequest",
+            "description": "Request body with feature names.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/BatchFeatureRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/BatchFeatureStatus"
+            }
+          },
+          "default": {
+            "description": "An error response received from the provider.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseModel"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Features_AccountGet": {
+            "$ref": "./examples/Features_AccountGet.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Purview/accounts/{accountName}/ingestionPrivateEndpointConnections": {
+      "get": {
+        "tags": [
+          "IngestionPrivateEndpoint"
+        ],
+        "summary": "Lists all ingestion private endpoint connections.",
+        "description": "Lists all ingestion private endpoint connections",
+        "operationId": "IngestionPrivateEndpointConnections_List",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/accountName"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/PrivateEndpointConnectionList"
+            }
+          },
+          "default": {
+            "description": "An error response received from the provider.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseModel"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "IngestionPrivateEndpointConnections_List": {
+            "$ref": "./examples/IngestionPrivateEndpointConnections_List.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Purview/accounts/{accountName}/ingestionPrivateEndpointConnectionStatus": {
```

Comment:
>ingestionPrivateEndpointConnectionStatus [](http://example.com/codeflow?start=123&length=40)

Can you please update the naming to reflect the action that is being taken ?

Can you also please explain why the update needs be done via a POST rather then using a standard REST resource with PUT\GET\DEETE  methods ?

Reply:
We are performing "Update Ingestion Private Endpoint Connection Status" operation using the POST API.

`IngestionPrivateEndpointConnection` is not an ARM resource (neither tracked, nor proxy), otherwise we would have used PUT / PATCH.

Reply:
Again the question is about modelling. Why have you not modelled it as a resource ? modelling as a resource is much more beneficial in the arm ecosystem rather than modelling it using POST actions ,many tools like arm templates etc donot supprt POST

Reply:
For purview customers, they only care about ingestion to work as expected. The internals of ingestion storage (which is in MSFT boundary) are not disclosed to customers. Hence, we want to keep it as a action.
Does not make sense to model it as a resource and manage it's lifecycle.

Reply:
Ok but please keep in mind that this means that this means that you will NOT be able to do this operations using ARM templates. Can you please get your manager or a senior member of your team to sign off on this comment ?

Reply:
Hi Suhas - I agree with your comments but for Purview, most interaction happens through UX and template deployments are not exercised. I am approving this just in the interest of unblocking the ignite release. Changes in this area will entail a lot of rework across a lot of teams.
Based on the feedback for this preview versions - we will take a call to take the rework post ignite.